src/glb_writer.rs

```diff
@@ -0,0 +1,561 @@
+    pub fn new() -> Self {
+        Self {
+            buffer_view: 0,
+            component_type: 0,
+            min: None,
+            max: None
+        }
+    }
+}
+
+struct Chunk {
+    chunk_length: u32,
+    padding_length: u32
+}
+
+impl Chunk {
+    pub fn new() -> Self {
+        Self {
+            chunk_length: 0,
+            padding_length: 0,
+        }
+    }
+
+    pub fn data_length(&self) -> u32 {
+        self.chunk_length + self.padding_length
+    }
+
+    pub fn total_length(&self) -> u32 {
+        GLTF_CHUNK_HEADER_LENGTH + self.chunk_length + self.padding_length
+    }
+}
+
+pub struct GlbWriter {
+    /// Number of points in this glTF point cloud
+    point_count: u32,
+    /// Total length of binary glTF, including header and all chunks
+    total_length: u32,
+    json_chunk: Chunk,
+    binary_chunk: Chunk,
+    buffer_length: u32,
+    accessor_position: Accessor,
+    accessor_color: Accessor,
+    accessor_feature_ids: Accessor,
+    buffer_view_position: BufferView,
+    buffer_view_color: BufferView,
+    buffer_view_feature_ids: BufferView,
+    buffer_view_iterations: BufferView,
+    buffer_view_point_id: BufferView,
+    buffer_view_last_xforms: BufferView,
+    buffer_view_last_color_xforms: BufferView,
+    json: String,
+}
+
+impl GlbWriter {
+    pub fn new() -> Self {
+        Self {
+            point_count: 0,
+            total_length: 0,
+            json_chunk: Chunk::new(),
+            binary_chunk: Chunk::new(),
+            buffer_length: 0,
+            accessor_position: Accessor::new(),
+            accessor_color: Accessor::new(),
+            accessor_feature_ids: Accessor::new(),
+            buffer_view_position: BufferView::new(),
+            buffer_view_color: BufferView::new(),
+            buffer_view_feature_ids: BufferView::new(),
+            buffer_view_iterations: BufferView::new(),
+            buffer_view_point_id: BufferView::new(),
+            buffer_view_last_xforms: BufferView::new(),
+            buffer_view_last_color_xforms: BufferView::new(),
+            json: String::new(),
+        }
+    }
+
+    pub fn write(&mut self, fname: &str, buffer: &Vec<OutputPoint>) {
+        self.compute_layout(&buffer);
+        self.make_json();
+
+        // Now that we have both chunks, update the total length
+        self.total_length = 
+            GLTF_HEADER_LENGTH + 
+            self.json_chunk.total_length() + 
+            self.binary_chunk.total_length();
+
+        let error_msg = format!("Cannot open {}", fname);
+        let mut file = File::create(fname).expect(&error_msg);
+        self.write_header(&mut file);
+        self.write_json_chunk(&mut file, &self.json);
+        self.write_binary_chunk(&mut file, buffer);
+    }
+
+    fn compute_layout(&mut self, buffer: &Vec<OutputPoint>) {
+        let point_count = buffer.len() as u32;
+        self.point_count = point_count;
+
+        // vec3 position
+        let position_length = point_count * SIZE_VEC3;
```

Comment:
furthermore, this layout code has a lot of boilerplate, it would be better to iterate over a Vec of buffer views and update them that way.